xskillscore/tests/test_deterministic.py

```diff
@@ -311,3 +315,17 @@ def test_pearson_r_integer():
     """Test whether arrays as integers work."""
     da = xr.DataArray([0, 1, 2], dims=['time'])
     assert pearson_r(da, da, dim='time') == 1
+
+
+@pytest.mark.parametrize('metrics', correlation_metrics + distance_metrics)
+@pytest.mark.parametrize('keep_attrs', [True, False])
+def test_keep_attrs(a, b, metrics, keep_attrs):
+    """Test keep_attrs for all metrics."""
+    metric, _metric = metrics
+    res = metric(a, b, 'time', keep_attrs=keep_attrs)
+    if keep_attrs:
+        assert res.attrs == a.attrs
```

Comment:
Maybe comment here that we're testing `a` since that's what `xr.apply_ufunc` keeps. The caveat here is that if you're working with correlating two diff datasets, say temperature and salinity, you'll only keep the metadata from the temperature. But I imagine you use `keep_attrs=True` if `a` and `b` have similar attributes.